File: examples/vbnet/03_JsonStore.vb
Imports StackExchange.Redis
Imports System

Module Program
    Sub Main(args As String())
        Dim redis = ConnectionMultiplexer.Connect("localhost")
        Dim db = redis.GetDatabase()

        Console.WriteLine("--- 03 JSON Document Store ---")

        Dim json As String = "{ ""name"": ""VB.NET"", ""type"": ""language"" }"

        ' JSON.SET
        Console.WriteLine("Storing JSON...")
        db.Execute("JSON.SET", "lang:vb", json)

        ' JSON.GET
        Console.WriteLine("Retrieving JSON...")
        Dim data = db.Execute("JSON.GET", "lang:vb", ".")
        Console.WriteLine($"Result: {data}")
    End Sub
End Module

File: examples/go/03_json_store.go
package main

import (
	"context"
	"fmt"

	"github.com/redis/go-redis/v9"
)

var ctx = context.Background()

func main() {
	rdb := redis.NewClient(&redis.Options{
		Addr: "localhost:6379",
	})

	fmt.Println("--- 03 JSON Document Store ---")

	jsonData := `{"name": "Gopher", "type": "mascot", "skills": ["concurrency"]}`

	// JSON.SET
	fmt.Println("Storing JSON...")
	rdb.Do(ctx, "JSON.SET", "mascot:1", jsonData)

	// JSON.GET
	fmt.Println("Retrieving JSON...")
	val, err := rdb.Do(ctx, "JSON.GET", "mascot:1", ".").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println("Data:", val)
}
